manifests/params.pp

```diff
@@ -6,71 +6,49 @@
+        $service_file_mode     = '0755'
+        $service_lockfile      = '/var/lock/subsys/jira'
+        $service_provider      = undef
+      } elsif versioncmp($facts['os']['release']['major'], '7') >= 0 {
+        $json_packages         = [ 'rubygem-json' ]
         $service_file_location = '/usr/lib/systemd/system/jira.service'
         $service_file_template = 'jira/jira.service.erb'
-        $service_lockfile = '/var/lock/subsys/jira'
-        $service_provider = 'systemd'
-      } elsif versioncmp($facts['operatingsystemmajrelease'], '6') >= 0 {
-        $json_packages = [ 'rubygem-json', 'ruby-json' ]
+        $service_file_mode     = '0644'
+        $service_lockfile      = '/var/lock/subsys/jira'
+        $service_provider      = 'systemd'
+      } elsif versioncmp($facts['os']['release']['major'], '6') >= 0 {
+        $json_packages         = [ 'rubygem-json', 'ruby-json' ]
         $service_file_location = '/etc/init.d/jira'
         $service_file_template = 'jira/jira.initscript.erb'
-        $service_lockfile = '/var/lock/subsys/jira'
-        $service_provider = undef
+        $service_file_mode     = '0755'
+        $service_lockfile      = '/var/lock/subsys/jira'
+        $service_provider      = undef
       } else {
         fail("\"${module_name}\" provides no service parameters
-            for \"${::osfamily}\" - \"${$facts['operatingsystemmajrelease']}\"")
+            for \"${facts['os']['family']}\" - \"${$facts['os']['release']['major']}\"")
       }
     }
     /Debian/: {
-      case $::operatingsystem {
-        'Ubuntu': {
-          if versioncmp($facts['operatingsystemmajrelease'], '15.04') >= 0 {
-            $json_packages = [ 'ruby-json' ]
-            $service_file_location = '/lib/systemd/system/jira.service'
-            $service_file_template = 'jira/jira.service.erb'
-            $service_lockfile = '/var/lock/subsys/jira'
-            $service_provider = 'systemd'
-          } else {
-            $json_packages = [ 'rubygem-json', 'ruby-json' ]
-            $service_file_location = '/etc/init.d/jira'
-            $service_file_template = 'jira/jira.initscript.erb'
-            $service_lockfile = '/var/lock/jira'
-            $service_provider = 'debian'
-          }
-        }
-        default: {
-          if versioncmp($facts['operatingsystemmajrelease'], '8') >= 0 {
-            $json_packages = [ 'ruby-json' ]
-            $service_file_location = '/lib/systemd/system/jira.service'
-            $service_file_template = 'jira/jira.service.erb'
-            $service_lockfile = '/var/lock/subsys/jira'
-            $service_provider = 'systemd'
-          } else {
-            $json_packages = [ 'rubygem-json', 'ruby-json' ]
-            $service_file_location = '/etc/init.d/jira'
-            $service_file_template = 'jira/jira.initscript.erb'
-            $service_lockfile = '/var/lock/jira'
-            $service_provider = 'debian'
-          }
-        }
-      }
```

Comment:
This drops support for Debian 7 and Ubuntu 14. That's fine, but it's probably good to do that in a separate PR first so it's included in the changelog.

Reply:
We already dropped support for Debian 7. I just picked a commit from @bastelfreak and did some cleanup:
* https://github.com/voxpupuli/puppet-jira/pull/320/commits/b216e8d77260b0aa21d6c724c94d887fc3bb2329